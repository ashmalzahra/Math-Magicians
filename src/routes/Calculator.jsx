import Calculator from '../components/calculator';

const Calc = () => (
  <div>
    <h1>Math Magicians</h1>
    <h3>Lets do some math!</h3>
    <section className="App">
      <Calculator />
    </section>
  </div>
);

export default Calc;
