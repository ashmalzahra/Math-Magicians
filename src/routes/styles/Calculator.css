.calc-text {
  font-size: xx-large;
  padding-left: 1.5%;
  margin-bottom: 0;
}
