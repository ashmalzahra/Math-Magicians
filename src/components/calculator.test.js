import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from './calculator';

describe('Checking page deployment', () => {
  test('Calculator component matches snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Calculator component', () => {
    render(<Calculator />);

    expect(screen.getByRole('table')).toBeInTheDocument();
    expect(screen.getByText(/AC/)).toBeInTheDocument();
  });
});
