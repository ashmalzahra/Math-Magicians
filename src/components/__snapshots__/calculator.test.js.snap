// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking page deployment Calculator component matches snapshot 1`] = `
<div
  id="container"
>
  <table>
    <tbody>
      <tr
        className="display-values"
      >
        <th>
          0
        </th>
      </tr>
      <tr
        className="row"
      >
        <td>
          <button
            className="grey"
            name="AC"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
           
        </td>
        <td>
          <button
            className="grey"
            name="+/-"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="%"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
        </td>
        <td>
          <button
            className="orange"
            name="รท"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <button
            className="grey"
            name="7"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="8"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="9"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
        </td>
        <td>
          <button
            className="orange"
            name="x"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <button
            className="grey"
            name="4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="5"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="6"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
        </td>
        <td>
          <button
            className="orange"
            name="-"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <button
            className="grey"
            name="1"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="2"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="3"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
        </td>
        <td>
          <button
            className="orange"
            name="+"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </td>
      </tr>
      <tr
        className="row"
      >
        <td
          colSpan={2}
        >
          <button
            className="grey"
            id="last"
            name="0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
        </td>
        <td>
          <button
            className="grey"
            name="."
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </td>
        <td>
          <button
            className="orange"
            name="="
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
